src/OrchardCore/OrchardCore.Search.Lucene.Core/QueryProviders/BooleanQueryProvider.cs

```diff
@@ -80,42 +80,50 @@ public Query CreateQuery(ILuceneQueryService builder, LuceneQueryContext context
 
     private Query CreateFilteredQuery(ILuceneQueryService builder, LuceneQueryContext context, Query query, JsonNode filter)
     {
-        Query filteredQuery = null;
+        Query filteredQuery;
 
         switch (filter.GetValueKind())
         {
             case JsonValueKind.Object:
-                var first = filter.AsObject().First();
+                var first = filter.AsObject().FirstOrDefault();
+                if (string.IsNullOrEmpty(first.Key))
```

Comment:
`first` can be null still. You are checking `.Key` which was not used before ...
You have wanted to write `if (first is null)`

Reply:
The default is an empty key value pair, not null.